gtsfm/loader/argoverse_dataset_loader.py

```diff
@@ -0,0 +1,132 @@
+""" Loader that reads images from Argoverse directories on disk.
+
+Authors: John Lambert
+"""
+
+import glob
+import os
+from pathlib import Path
+from typing import Optional
+
+import numpy as np
+from gtsam import Cal3Bundler, Point3, Pose3, Rot3
+
+from argoverse.data_loading.simple_track_dataloader import SimpleArgoverseTrackingDataLoader
+from argoverse.utils.calibration import get_calibration_config
+from argoverse.utils.se3 import SE3
+
+import gtsfm.utils.io as io_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+
+class ArgoverseDatasetLoader(LoaderBase):
+    """ """
+
+    def __init__(self, dataset_dir: str, log_id: str, stride: int = 10, max_num_imgs: int = 20) -> None:
+        """
+        Args:
+            dataset_dir: directory where raw Argoverse logs are stored on disk
+            log_id: unique ID of vehicle log
+            stride: sampling rate, e.g. every 2 images, every 4 images, etc.
+            max_num_imgs: number of images to load (starting from beginning of log sequence)
+        """
```

Comment:
make make all variables private?

Reply:
done, thanks

Reply:
Underscore should come as a prefix?

https://docs.python.org/3/tutorial/classes.html#private-variables

Reply:
i dont think we're following "single underscore prefix in private class variables" anywhere else in gtsfm though -- is it something we want to switch to now?

Reply:
was using the underscore suffix,  which i guess i more appropriate for c++

Reply:
I have been trying to use it as different places: 
1. https://github.com/borglab/gtsfm/blob/master/gtsfm/averaging/rotation/shonan.py
2. https://github.com/borglab/gtsfm/blob/master/gtsfm/loader/yfcc_imb_loader.py

Although I am not sure if this is the way to go and I am happy to settle on a common convention.

Reply:
@ayushbaid thanks for sharing those links. I switched to underscore prefix, I'm fine with that